[FSDP] Enhance test for checkpoint

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #75236
* #75227

Make sure that the param names are such as we can load from checkpoint
into a local model.

Differential Revision: [D35377774](https://our.internmc.facebook.com/intern/diff/D35377774/)

rohan-varma committed:
[FSDP] Enhance test for checkpoint

Make sure that the param names are such as we can load from checkpoint
into a local model.

Differential Revision: [D35377774](https://our.internmc.facebook.com/intern/diff/D35377774/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[FSDP] Enhance test for checkpoint"


Make sure that the param names are such as we can load from checkpoint
into a local model.

Differential Revision: [D35377774](https://our.internmc.facebook.com/intern/diff/D35377774/)

[ghstack-poisoned]

awgu commented on test/distributed/fsdp/test_fsdp_state_dict.py:
```diff
@@ -324,12 +326,30 @@ def _create_module():
+            for (p1, p2) in zip(fsdp.parameters(), local.parameters()):
+                self.assertNotEqual(p1, p2)
+
+        local.load_state_dict(deepcopy(state_dict))
```
Curious, what is the need for the `deepcopy()`?

rohan-varma replied:
It's used in a few other tests in this file, and I think the aim is to better simulate a scenario where we don't have any access to the original state_dict and instead recover it from a file. Though I think doing an actual torch.save / torch.load using tempfiles would a better test here.

awgu reviewed (approved):
Looks good!

rohan-varma commented:
@pytorchbot merge this please

rohan-varma committed:
Update on "[FSDP] Enhance test for checkpoint"


Make sure that the param names are such as we can load from checkpoint
into a local model.

Differential Revision: [D35377774](https://our.internmc.facebook.com/intern/diff/D35377774/)

[ghstack-poisoned]

rohan-varma commented:
@pytorchbot merge this please
